lib/base-compiler.ts

```diff
@@ -900,13 +944,91 @@ export class BaseCompiler implements ICompiler {
         );
     }
 
+    getDefaultOrOverridenToolchainPath(overrides: ConfiguredOverrides): string {
+        for (const override of overrides) {
+            if (override.value) {
+                const possible = this.compiler.possibleOverrides?.find(ov => ov.name === override.name);
+                if (possible && possible.name === CompilerOverrideType.toolchain) {
+                    return override.value;
+                }
+            }
+        }
+
+        return this.toolchainPath;
+    }
+
+    getOverridenToolchainPath(overrides: ConfiguredOverrides): string | false {
+        for (const override of overrides) {
+            if (override.value) {
+                const possible = this.compiler.possibleOverrides?.find(ov => ov.name === override.name);
+                if (possible && possible.name === CompilerOverrideType.toolchain) {
+                    return override.value;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    changeOptionsBasedOnOverrides(options: string[], overrides: ConfiguredOverrides): string[] {
+        let sysrootPath: string | undefined;
+        const overriddenToolchainPath = this.getOverridenToolchainPath(overrides);
+        if (overriddenToolchainPath) {
```

Comment:
I think this could be:
```ts
const sysrootPath = overriddenToolchainPath ? getSysrootByToolchainPath(overriddenToolchainPath) : undefined;
```

or there might be some cool `?.` or `??` magic typescripty way to do it. Just to avoid the `let`/const and whatnot